projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/time_based_queue.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import time
+from collections import deque
+from dataclasses import dataclass
+from random import randint
+from typing import Any
+
+
+@dataclass
+class _QueueEntry:
+    element: Any
+    ready_time: float
+
+
+class TimeBasedQueue:
```

Comment:
I think 99% of this code can be deleted if you use a priorityQueue from the standard python library https://docs.python.org/3/library/queue.html#queue.PriorityQueue?

Reply:
That is NOT a priority queue. The point of this queue is to delay dequeu operations until they are ready.